airflow/providers/amazon/aws/hooks/cloud_formation.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+This module contains AWS CloudFormation Hook
+"""
+from botocore.exceptions import ClientError
+
+from airflow.contrib.hooks.aws_hook import AwsHook
+
+
+class AWSCloudFormationHook(AwsHook):
+    """
+    Interact with AWS CloudFormation.
+    """
+
+    def __init__(self, region_name=None, *args, **kwargs):
+        self.region_name = region_name
+        self.conn = None
+        super().__init__(*args, **kwargs)
+
+    def get_conn(self):
+        if not self.conn:
+            self.conn = self.get_client_type('cloudformation', self.region_name)
+        return self.conn
+
+    def get_stack_status(self, stack_name):
+        """
+        Get stack status from CloudFormation.
+        """
+        cloudformation = self.get_conn()
+
+        self.log.info('Poking for stack %s', stack_name)
+
+        try:
+            stacks = cloudformation.describe_stacks(StackName=stack_name)['Stacks']
+            return stacks[0]['StackStatus']
+        except ClientError as e:
+            if 'does not exist' in str(e):
+                return None
+            else:
+                raise e
```

Comment:
```python
        except ClientError as e:
            if 'does not exist' in str(e):
                return None
            raise e
```

I am also wondering if `ClientError` is the correct exception type to catch.

The docs https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Client.describe_stacks are saying that it raises an `AmazonCloudFormationException` when it does not exist.

So then I would only catch this:
```python
        except AmazonCloudFormationException as e:
            return None
```

Reply:
Yep, I read that, however there is no class by name AmazonCloudFormationException in boto.

I could add:
```
e.response['Error']['Code'] == 'AmazonCloudFormationException'
```
as suggested [here](https://stackoverflow.com/a/47040476/2489287)

However since we are using boto's cloudformation client here it is certain that that would be the value of the error code in the exception.

Should I make this change? or should we go for the brevity of simply catching `ClientError` and not check its error code value.

Reply:
In my opinion going for the error code is far better than checking the message string.

Reply:
Ah you mean we still need to check the `if 'does not exist' in str(e):` ?

Reply:
Hm ya then let's simply stick with it - so now change needed from your side :)

Thanks for the explanation. 👍

Reply:
We could check the error code instead of the message string, and assume that `ValidationError` returned from `describe_stacks` could only be due to stack not existing, however potentially there could be other scenarios in which that can happen.

![image](https://user-images.githubusercontent.com/5708714/73180571-39a28d00-411e-11ea-960a-69a582d1b7e0.png)

Reply:
Yes, so let's better check the message string then.